packages/minter-contracts/ligo/src/english_auction/english_auction_tez.mligo

```diff
@@ -228,35 +244,90 @@ let cancel_auction(asset_id, storage : nat * storage) : return = begin
     assert_msg (auction_in_progress(auction), "NOT_IN_PROGRESS");
-    assert_msg(Tezos.sender <> auction.seller, "SEllER_CANT_BID");
-    assert_msg(Tezos.sender <> auction.highest_bidder, "NO_SELF_OUTBIDS");
-    (if not valid_bid_amount(auction)
+    assert_msg(bidder <> auction.seller, "SEllER_CANT_BID");
+    assert_msg(bidder <> auction.highest_bidder, "NO_SELF_OUTBIDS");
+    (if not valid_bid_amount(auction, bid_amount)
       then ([%Michelson ({| { FAILWITH } |} : string * (tez * tez * address * timestamp * timestamp) -> unit)] ("INVALID_BID_AMOUNT", (auction.current_bid, Tezos.amount, auction.highest_bidder, auction.last_bid_time, Tezos.now)) : unit)
       else ());
-    let op_list : operation list = if first_bid(auction) then
+    let op_list : operation list = if dont_return_bid(auction) then
       ([] : operation list) else
       let return_bid : operation = transfer_tez(auction.current_bid,  auction.highest_bidder) in
       [return_bid] in
     let new_end_time = if auction.end_time - Tezos.now <= auction.extend_time then
       Tezos.now + auction.extend_time else auction.end_time in
-    let updated_auction_data = {auction with current_bid = Tezos.amount; highest_bidder = Tezos.sender; last_bid_time = Tezos.now; end_time = new_end_time;} in
+    let updated_auction_data = {auction with current_bid = bid_amount; highest_bidder = bidder; 
+                                last_bid_time = Tezos.now; end_time = new_end_time;
+#if OFFCHAIN_BID
+                                last_bid_offchain = is_offchain; 
+#endif
+                               } in
     let updated_auctions = Big_map.update asset_id (Some updated_auction_data) storage.auctions in
     (op_list , {storage with auctions = updated_auctions})
   end
 
+let place_bid_onchain(asset_id, storage : nat * storage) : return = begin
+    let bid_amount = Tezos.amount in 
+    let bidder = Tezos.sender in 
+    let auction : auction = get_auction_data(asset_id, storage) in 
+    let (ops, new_storage) = place_bid(asset_id, auction, bid_amount, bidder, storage
+#if OFFCHAIN_BID
+      , false
+#endif
+      ) in 
+    (ops, new_storage)
+  end
+
+#if OFFCHAIN_BID
+let place_bid_offchain(offchain_bid_data, bidder, storage : offchain_bid_data * address * storage) : return = begin
+    let { asset_id = asset_id;
+          bid_amount = bid_amount;
+        } = offchain_bid_data in 
+    let auction : auction = get_auction_data(asset_id, storage) in
+    let (ops, new_storage) = place_bid(asset_id, auction, bid_amount, bidder, storage, true ) in 
+    (ops, new_storage)
+  end
+
+let bid_with_permit (p, storage : permit_bid_param * storage)  : return = begin 
+    fail_if_not_admin(storage.admin);
+    let {offchain_bid_data = offchain_bid_data;
+         permit = permit; } = p in 
+    let param_hash = Crypto.blake2b (Bytes.pack offchain_bid_data) in 
+    let v : unit = check_permit (permit, 0n, param_hash) in  (*Always set counter to 0*)
```

Comment:
why is the counter always set to `0`?

Reply:
Since admin is always submitting in this case, and because bids are strictly increasing, we are not worried about replay attacks. See: https://github.com/tqtezos/minter-sdk/blob/main/packages/minter-contracts/ligo/src/fixed_price_sale/fixed_price_sale_market_offchain_nocounter.mligo

Reply:
ok, please add documentation to that effect as we discussed